packages/contracts/src/contracts/current/protocol/AssetProxy/proxies/ERC20Proxy.sol

```diff
@@ -29,18 +30,61 @@ contract ERC20Proxy is
     IAssetProxy
 {
 
-    /// @dev Transfers ERC20 tokens. Either succeeds or throws.
+    /// @dev Transfers ERC20 assets. Either succeeds or throws.
     /// @param assetMetadata ERC20-encoded byte array.
-    /// @param from Address to transfer token from.
-    /// @param to Address to transfer token to.
-    /// @param amount Amount of token to transfer.
+    /// @param from Address to transfer asset from.
+    /// @param to Address to transfer asset to.
+    /// @param amount Amount of asset to transfer.
     function transferFrom(
         bytes assetMetadata,
         address from,
         address to,
         uint256 amount)
         external
         onlyAuthorized
+    {
+        transferFromInternal(
+            assetMetadata,
+            from,
+            to,
+            amount
+        );
+    }
+
+    /// @dev Makes multiple transfers of assets. Either succeeds or throws.
+    /// @param assetMetadata Array of byte arrays encoded for the respective asset proxy.
+    /// @param from Array of addresses to transfer assets from.
+    /// @param to Array of addresses to transfer assets to.
+    /// @param amounts Array of amounts of assets to transfer.
+    function batchTransferFrom(
+        bytes[] memory assetMetadata,
```

Comment:
Would an overloaded for ERC20 proxy with a single `assetMetadata` make sense in the future? In the example of the marketSell/Buy this data should always be the same or does this break our duck typing when using proxies?

Reply:
Potentially. I don't think this will end up being too costly since we aren't repeating the `assetMetadata` a ton in any calldata, though.